File: Source/FlatShare 2.0/FlatshareCentral/src/main/java/com/joinflatshare/ui/faq/FaqActivity.java
package com.joinflatshare.ui.faq;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

import com.joinflatshare.FlatshareCentral.databinding.ActivityFaqBinding;
import com.joinflatshare.pojo.faq.FaqItem;
import com.joinflatshare.ui.base.BaseActivity;
import com.joinflatshare.utils.mixpanel.MixpanelUtils;

import java.util.ArrayList;

public class FaqActivity extends BaseActivity {
    private ActivityFaqBinding viewBind;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewBind = ActivityFaqBinding.inflate(getLayoutInflater());
        setContentView(viewBind.getRoot());
        showTopBar(this, true, "Settings", 0, 0);
        new FaqViewBind(this, viewBind);
        new FaqListener(this, viewBind);
        getFaqs();
        MixpanelUtils.INSTANCE.onButtonClicked("Settings");
    }

    private void getFaqs() {
        ArrayList<FaqItem> faqItems = new ArrayList<>();
        FaqItem item = new FaqItem();
        item.setQuestion("What is flatshare?");
        item.getAnswers().add("flatshare is a new app for students and young adults living in India where members connect with each other for dating, hunting and sharing a flat together.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("Why did you create flatshare?");
        item.getAnswers().add("Who you live with = Who you become.");
        item.getAnswers().add("People you surround yourself with can have a huge impact on your life.");
        item.getAnswers().add("That's why we created flatshare - to make it super easy for you to connect with other better and interesting people, people just like you.");
        item.getAnswers().add("People who get you.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("What is Flathunt Together and how it’s different from Shared Flat Search?");
        item.getAnswers().add("Flathunt together allows you to connect with your future flatmate before you have a flat. Once you’ve found a flatmate you can hunt a private flat and move-in together.");
        item.getAnswers().add("If you are looking for an individual bed/room in an already rented flat.\nShared Flat Search is a better option for you.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("How do I find a shared flat?");
        item.getAnswers().add("Looking for an individual bed/room in an already rented flat.\nFinding flatshare has never been easier.");
        item.getAnswers().add(getEmojiByUnicode(0x1F4AA) + " Build your profile: Your profile is a snapshot of who you are.");
        item.getAnswers().add(getEmojiByUnicode(0x2705) + " Check Flats: Based on your preferences, we’ll show you flats that you might like to check.");
        item.getAnswers().add(getEmojiByUnicode(0x1F91D) + " Match and Chat: If the checking is mutual, you can chat with the flatmates and decide if it’s the right flat for you.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("How do I find a flatmate for my already rented flat?");
        item.getAnswers().add("Got a vacant room or bed?\nWith flatshare, you can find flatmates like friends.");
        item.getAnswers().add(getEmojiByUnicode(0x1F3E0) + " Add your flat: Add your flat by entering your flatname.");
        item.getAnswers().add(getEmojiByUnicode(0x1F4AA) + " Build your flat profile: Your flat profile includes you, your flatmates, flat photos and other details.");
        item.getAnswers().add(getEmojiByUnicode(0x2705) + " Check potential flatmates: Based on your details and preferences, we’ll show you potential flatmates who you might like to live with.");
        item.getAnswers().add(getEmojiByUnicode(0x1F91D) + "  Match and Chat: If the checking is mutual, you can chat with them and decide if they are the right flatmate for you.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("Can I send a chat request directly and skip the mutual checking?");
        item.getAnswers().add("Yes, you can send chat request and skip the mutual checking.\n" +
                "Your chat request still has to be accepted for you to connect.");
        faqItems.add(item);


        item = new FaqItem();
        item.setQuestion("Why are there daily limits on Views, Checks and Chat Requests?");
        item.getAnswers().add("Our algorithm gives you only the Top Recommendations based on your profile and preferences so that you don't need to find a needle in a haystack. We do it for you.");
        item.getAnswers().add("Limits help everyone send and receive the Checks and Chat Requests to the ones they are most interested in.\n" +
                "Everyone gets much better quality of Checks and Chat Requests.");
        faqItems.add(item);


        item = new FaqItem();
        item.setQuestion("What does f - FlatScore mean?");
        item.getAnswers().add("FlatScore is a measure of community trust. \nHigher the community trusts you, the higher your score. \n" +
                "No spammer, scammer, or fraudster can have a high FlatScore.");
        item.getAnswers().add("FlatScore helps you in connecting with people who have FlatScore similar to you and keep spammers, scammers, & fraudsters out of flatshare.");
        faqItems.add(item);


        item = new FaqItem();
        item.setQuestion("How do Invites work?");
        item.getAnswers().add("Everybody who joins flatshare can invite a few of their friends.\n" +
                "If your friends join you get to invite even more friends.");
        item.getAnswers().add("All the invites expire in 48 hours, so make sure your friends know you’ve invited them.");
        item.getAnswers().add("You’ll get credit for the invite on their profile and 1000 flatcoins for each successful invite.");
        faqItems.add(item);


        item = new FaqItem();
        item.setQuestion("When should I report someone?");
        item.getAnswers().add("We care deeply about our users and strive to make our community safe and comfortable. We will not tolerate any form of abuse or harassment.");
        item.getAnswers().add("If you see any member trying to impersonate someone else, asking for brokerage, sending inappropriate messages or photos, go to the bottom of their profile and report.");
        item.getAnswers().add("You can also reach out directly to our team directly at hello@flatshare.club and share your feedback. We would love to hear from you.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("I signed up with the wrong name/birthdate/gender. What should I do?");
        item.getAnswers().add("If you signed up with the wrong name, birthdate or gender, reach out to our team directly at hello@flatshare.club along with your mobile number and aadhar card and we’ll change it for you.");
        faqItems.add(item);

        item = new FaqItem();
        item.setQuestion("Can I share feedback or a feature request?");
        item.getAnswers().add("Constant evolution is our MO. We're always looking for ways to improve and make flatshare the best it can be. So don't be shy - let us know what you think!");
        item.getAnswers().add("If you have any other questions, feedback or want to report a negative experience, please reach out at hello@flatshare.club");
        faqItems.add(item);
        viewBind.expProfileFaq.setVisibility(View.VISIBLE);
        viewBind.expProfileFaq.setAdapter(new FaqAdapter(this, faqItems));
    }

    protected void collapseGroup(int position) {
        if (viewBind.expProfileFaq.isGroupExpanded(position))
            viewBind.expProfileFaq.collapseGroup(position);
    }
}

File: Flatshare Code/FlatShare 2.0/FlatshareCentral/src/main/java/com/joinflatshare/ui/faq/FaqActivity.kt
package com.joinflatshare.ui.faq

import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.joinflatshare.FlatshareCentral.databinding.ActivityFaqBinding
import com.joinflatshare.pojo.faq.FaqItem
import com.joinflatshare.ui.base.BaseActivity
import com.joinflatshare.utils.mixpanel.MixpanelUtils

/**
 * Created by debopam on 21/05/24
 */
class FaqActivity : BaseActivity() {
    private lateinit var viewBind: ActivityFaqBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewBind = ActivityFaqBinding.inflate(layoutInflater)
        setContentView(viewBind.root)
        showTopBar(this, true, "FAQ", 0, 0)
        MixpanelUtils.onButtonClicked("FAQ")
        init()
    }

    private fun init() {
        viewBind.rvFaq.layoutManager = LinearLayoutManager(this)
        viewBind.rvFaq.adapter = FaqAdapter(getFaqs())
    }

    private fun getFaqs(): ArrayList<FaqItem> {
        val faqItems = ArrayList<FaqItem>()
        faqItems.add(
            FaqItem(
                "Why did you create Flatshare?",
                true
            )
        )
        faqItems.add(
            FaqItem(
                "Who you live with = Who you'll become.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "People you surround yourself with can have a huge impact on your life.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "That's why we created flatshare - to make your flat and flatmate search as simple and fun as possible, without you paying any brokerage.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "So that you can LIVE YOUR BEST LIFE.",
                false
            )
        )


        faqItems.add(
            FaqItem(
                "How does Flatshare help with finding your Perfect Flat & Flatmate?",
                true
            )
        )
        faqItems.add(
            FaqItem(
                "Flatshare uses an intelligent matching algorithm that only gives you the top recommendations based on your profile and preferences.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "This eliminates the need to search through an overwhelming number of options.\n" +
                        "We do the work for you, making the process more efficient and tailored to your needs.",
                false
            )
        )

        faqItems.add(
            FaqItem(
                "What is FlatScore?",
                true
            )
        )
        faqItems.add(
            FaqItem(
                "Flatscore is a measure of trust within the Flatshare community.\n" +
                        "A higher Flatscore indicates a higher level of trust from the community. ",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "Spammers, scammers, and fraudsters are unlikely to have a high Flatscore.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "This feature helps you connect with people who have a similar Flatscore, ensuring a safe and trustworthy community.",
                false
            )
        )



        faqItems.add(
            FaqItem(
                "When should I report someone?",
                true
            )
        )

        faqItems.add(
            FaqItem(
                "We take the safety and comfort of our members seriously.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "Any form of abuse, harassment, impersonation, solicitation of brokerage fees, or inappropriate messages or photos should be reported. ",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "Go to the bottom of the person's profile and select 'report'.\n" +
                        "You can also reach out to our team directly at hello@flatshare.club to share your feedback.",
                false
            )
        )

        faqItems.add(
            FaqItem(
                "I signed up with the wrong name/birthdate/gender. What should I do?",
                true
            )
        )
        faqItems.add(
            FaqItem(
                "If you signed up with incorrect personal information, such as name, birth date, or gender, please reach out to our team at hello@flatshare.club with your mobile number and Aadhar card details.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "We’ll change it for you.",
                false
            )
        )



        faqItems.add(
            FaqItem(
                "Can I share feedback or a feature request?",
                true
            )
        )
        faqItems.add(
            FaqItem(
                "Absolutely! We value your feedback and are always striving to improve Flatshare.",
                false
            )
        )
        faqItems.add(
            FaqItem(
                "f you have any suggestions, feature requests, or feedback, please reach out to us at hello@flatshare.club. We'd love to hear from you.",
                false
            )
        )

        return faqItems
    }
}